src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/EditingTargetCodeExecution.java

```diff
@@ -158,6 +161,44 @@ public void executeRange(Range range)
       executeRange(range, null, false);
    }
    
+   public void sendSelectionToTerminal(boolean terminalExecuteWhenNotFocused)
+   {
+      if (terminalExecuteWhenNotFocused && !docDisplay_.isFocused())
+      {
+         // the current editor may be in another window; if one of our source
+         // windows was last focused, use that one instead
+         SourceWindowManager manager = 
```

Comment:
The SendToTerminalEvent is a CrossWindowEvent, as I am sending it from the satellite (if any) to the main window.

https://github.com/rstudio/rstudio/blob/c231f92ef4826379ad3e76212036e404f53a4643/src/gwt/src/org/rstudio/studio/client/workbench/views/terminal/events/SendToTerminalEvent.java#L25

For the other gunk, I was following the pattern of the send R to Console code (in terms of the second fake command, and the way it ultimately decides which editor's buffer to use):

https://github.com/rstudio/rstudio/blob/c231f92ef4826379ad3e76212036e404f53a4643/src/gwt/src/org/rstudio/studio/client/workbench/views/console/shell/Shell.java#L429-L430

Console is more complicated than my terminal scenario, in that it bounces the request from the editor, through the Shell (in case the Console wants to take the opportunity to do something different), then invokes that NonFocused command variant on the editor's window, which then can do the cross-window event.

For mine, if we're ok with the main menu only acting on the editor in the main menu (so you'd have to use the shortcuts to activate the command in the satellite window, but that's really the case, anyway), then I can get rid of this second command.